packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts

```diff
@@ -59,4 +60,155 @@ export interface IComboBoxProps extends ISelectableDroppableTextProps<IComboBox>
    * The IconProps to use for the button aspect of the combobox
    */
   buttonIconProps?: IIconProps;
+
+  /**
+   * Theme provided by HOC.
+   */
+  theme?: ITheme;
+
+  /**
+   * Custom styles for this component
+   */
+  styles?: IComboBoxStyles;
 }
+
+export interface IComboBoxStyles {
+  /**
+   * Style for the container which has the ComboBox and the label
+   */
+  container?: IStyle;
+
+  /**
+   * Style for the label element of the ComboBox.
+   */
+  label?: IStyle;
+
+  /**
+   * Base styles for the root element of all ComboBoxes.
+   */
+  root?: IStyle;
+
+  /**
+   * Styles for the root element for variant of ComboBox with an errorMessage in the props.
+   */
+  rootError?: IStyle;
+
+  /**
+ * Styles for variant of ComboBox where allowFreeForm is false in the props.
+ */
+  rootDisallowFreeForm?: IStyle;
+
+  /**
+ * Styles for when the ComboBox is hovered or focused. These styles are applied for all comboBoxes except when
+ * the comboBox is disabled.
+ */
+  rootHoveredOrFocused?: IStyle;
+
+  /**
+   * Styles for when the comboBox is disabled. These styles override all the other styles.
+   * NOTE : Hover (or) Focused (or) active styles are not applied for disabled comboBoxes.
+   */
+  rootDisabled?: IStyle;
+
+  /**
+   * Base styles for the input element - which contains the currently selected
+   * option.
+   */
+  input?: IStyle;
+
+  /**
+   * Style override for the input element when comboBox is disabled.
+   */
+  inputDisabled?: IStyle;
+
+  /**
+   * Base styles for the caretDown button when enabled.
+   */
+  caretDown?: IStyle;
+
+  /**
+* Style override for caretDown button when freeform is disallowed.
+*/
+  caretDownDisallowFreeForm?: IStyle;
+
+  /**
+  * Style override for caretDown button when disabled.
+  */
+  caretDownDisabled?: IStyle;
+
+  /**
+   * Style override for caretDown button when enabled/hovered.
+   */
+  caretDownHovered?: IStyle;
+
+  /**
+  * Style override for caretDown button when enabled/active.
+  */
+  caretDownActive?: IStyle;
+
+  /**
+   * Styles for the error Message text of the comboBox.
+   */
+  errorMessage?: IStyle;
+
+  /**
+   * Styles for the callout.
+   */
+  callout?: IStyle;

```

Comment:
Yes, but it would be nice to not need to override onRenderOption just for styling. That should only be required if you want to change the html structure